arrows/vxl/high_pass_filter.cxx

```diff
@@ -0,0 +1,477 @@
+                        unsigned kernel_height )
+  {
+    if( treat_as_interlaced )
+    {
+      // if interlaced, split the image into odd and even views
+      // transpose all input and ouput images so that the horizontal
+      // smoothing function produces vertical smoothing
+      vil_image_view< PixType > im_even_t =
+        vil_transpose( even_rows( src ) );
+      vil_image_view< PixType > im_odd_t =
+        vil_transpose( odd_rows( src ) );
+
+      vil_image_view< PixType > smooth_even_t =
+        vil_transpose( even_rows( dst ) );
+      vil_image_view< PixType > smooth_odd_t =
+        vil_transpose( odd_rows( dst ) );
+
+      // use a half size odd kernel since images are half height
+      int half_kernel_height = kernel_height / 2;
+      if( half_kernel_height % 2 == 0 )
+      {
+        ++half_kernel_height;
+      }
+      // Smooth transposed inputs with the horizontal smoothing
+      box_average_horizontal( im_even_t, smooth_even_t, half_kernel_height );
+      box_average_horizontal( im_odd_t, smooth_odd_t, half_kernel_height );
+    }
+    else
+    {
+      // if not interlaced, transpose inputs and outputs and apply horizontal
+      // smoothing.
+      vil_image_view< PixType > grey_img_t = vil_transpose( src );
+      vil_image_view< PixType > smooth_t = vil_transpose( dst );
+      box_average_horizontal( grey_img_t, smooth_t, kernel_height );
+    }
+  }
+
+  // Apply a high pass filter to one frame and return the output
+  template < typename PixType >
+  kwiver::vital::image_container_sptr
+  filter( vil_image_view< PixType >& input )
+  {
+    vil_image_view< PixType > output;
+    switch( mode )
+    {
+      case MODE_box:
+        output = box_high_pass_filter( input );
+        break;
+      case MODE_bidir:
+        output = bidirection_box_filter( input );
+        break;
+      default:
+        return nullptr;
+    }
+
+    // Only report the summation of directional filterings, contained in the
+    // last plane
+    if( output_net_only && output.nplanes() > 1 )
+    {
+      output = vil_plane( output, output.nplanes() - 1 );
+    }
+
+    return std::make_shared< vxl::image_container >( output );
+  }
+};
+
+// ----------------------------------------------------------------------------
+high_pass_filter
+::high_pass_filter()
+  : d( new priv( this ) )
+{
+  attach_logger( "arrows.vxl.high_pass_filter" );
+}
+
+// ----------------------------------------------------------------------------
+high_pass_filter
+::~high_pass_filter()
+{
+}
+
+// ----------------------------------------------------------------------------
+vital::config_block_sptr
+high_pass_filter
+::get_configuration() const
+{
+  // get base config from base class
+  vital::config_block_sptr config = algorithm::get_configuration();
+
+  config->set_value( "mode", mode_converter().to_string(
+                       d->mode ),
+                     "Operating mode of this filter, possible values: box, bidir" );
```

Comment:
I missed this on the first comment, but there is a way to automatically insert the options into the help string.
```suggestion
                     "Operating mode of this filter, possible values: " 
                     + mode_converter().element_name_string()  );
```